test/functional/tests/io_class/test_io_class_pinning_eviction.py

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright(c) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import pytest
+from collections import namedtuple
+
+
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from api.cas.ioclass_config import IoClass, default_config_file_path
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from core.test_run import TestRun
+from test_utils.size import Size, Unit
+from .io_class_common import (
+    prepare,
+    mountpoint,
+    get_io_class_occupancy,
+    run_io_dir,
+)
+
+from api.cas import ioclass_config, casadm
+
+cache_size = Size(256, Unit.MB)
+core_size = Size(2, Unit.GiB)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_pinning_eviction():
+    """
+    title: IO class pinning eviction - stress.
+    description: |
+        Check Open CAS ability to prevent from eviction of pinned IoClass - stress test.
+    pass_criteria:
+        - IoClasses loaded successfully
+        - Pinned class occupies the same ammount of cache after each IO operation on different IoClass
+    """
+
+    with TestRun.step("Prepare devices"):
+        cache, core = prepare(core_size=core_size, cache_size=cache_size)
+
+    with TestRun.step("Mount filesystem"):
+        core.create_filesystem(Filesystem.xfs)
+        core.mount(mountpoint)
+
+    with TestRun.step("Prepare test dirs"):
+        IoclassConfig = namedtuple("IoclassConfig", "id eviction_prio max_occupancy dir_path")
+        io_classes = [
+            IoclassConfig(1, 0, 0.60, f"{mountpoint}/A"),
```

Comment:
Priority "0" is just a normal priority. To pin an ioclass priority must be an empty string. Please compare result of `casadm -C -L -i <cache_id>` for the two following ioclass configs:
```
cat ioclass-config.csv 
IO class id,IO class name,Eviction priority,Allocation
0,unclassified,,0.1
22,direct&done,20,1
```
```
cat ioclass-config.csv 
IO class id,IO class name,Eviction priority,Allocation
0,unclassified,0,0.1
22,direct&done,20,1
```